server/lib/findRoomByNonValidatedName.ts

```diff
@@ -0,0 +1,22 @@
+import s from 'underscore.string';
+
+import { getValidRoomName } from '../../app/utils/lib/getValidRoomName';
+import { Rooms } from '../../app/models/server';
+import type { IRoom } from '../../definition/IRoom';
+
+export const findRoomByNonValidatedName = function(name: string): IRoom | undefined {
+	const room = Rooms.findOneByName(name);
+	if (room) {
+		return room;
+	}
+
+	let channelName = s.trim(name);
+	try {
+		// TODO evaluate if this function call should be here
+		channelName = getValidRoomName(channelName, undefined, { allowDuplicates: true });
+	} catch (e) {
+		console.error(e);
+	}
+
+	return Rooms.findOneByName(channelName);
+};
```

Comment:
I dont know why you removed the logic from the model, but I would change this function to async, and use the raw model instead. and I know its not our default, but I would use the raw collection as the first parameter injecting the model and removing the import (or I would keep this one as a method of Room raw collection)